src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this.length === 1) {
+      return String(this[0]);
+    }
+
+    let string = this[0] === null ? '' : String(this[0]);
+
+    for (let i = 1; i < this.length; i++) {
+      let el = this[i];
+
+      if (el === null || el === undefined) {
+        el = '';
+      }
```

Comment:
let's add element to the result string only if it's not null or undefined instead of set empty string here